### PR TITLE
Added menu option to clear the map.

### PR DESCRIPTION
**Pull request by teamagam3**
Currently deletes any sent or received entities (without users loations).

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/79" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 1]
```diff
@@ -300,6 +300,33 @@ public void onGGMapViewReady() {
         registerForLocationUpdates();
     }
 
+    public void clearSentLocationsLayer() {
+        Entity[] entities = mSentLocationsLayer.getEntities().toArray(new Entity[0]);
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: you can change all for loops to for-each:
for (Entity entity : mReceivedLocationsLayer.getEntities()) {
            mReceivedLocationsLayer.removeEntity(entity.getId());
}

!fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 2]
```diff
@@ -300,6 +300,33 @@ public void onGGMapViewReady() {
...
+        for (int i = 0; i < entities.length; i++) {
+            if (shouldRemoveUserLocationEntity(entities[i].getId())) {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: misplaced. should be in clearUsersLocationLayer
!fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 3]
```diff
@@ -300,6 +300,33 @@ public void onGGMapViewReady() {
...
+                mSentLocationsLayer.removeEntity(entities[i].getId());
+            }
+        }
+    }
+
+    public void clearUsersLocationLayer() {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: I see now that we dont use this method. you can safely delete it.
!fix

File: app/src/debug/res/values/strings.xml [hunk 4]
```diff
@@ -103,6 +103,7 @@
     <string name="menu_setting_title">Settings</string>
     <string name="menu_setting_friends">Friends</string>
     <string name="menu_action_settings">Settings</string>
+    <string name="menu_action_clear_map">Clear Map</string>
```
Comment (teamagam1) [inline app/src/debug/res/values/strings.xml:106]: why add both debug string and main string. you can add only one to main/strings.xml

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 5]
```diff
@@ -300,6 +300,33 @@ public void onGGMapViewReady() {
         registerForLocationUpdates();
     }
 
+    public void clearSentLocationsLayer() {
```
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java:303]: consider adding method to vectorlayer: removeAllEntities()
!resolve

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 3 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !fixed

Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: I created this method for future use.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 1 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !fixed

Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: I can't, I delete the items inside the collection.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 2 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: It's in the right place, I'll rename the method

Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 5 above)
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java:303]: !resolved

**Pull request merged**